### PR TITLE
convertToSingleHost.js shows usage if no args given

### PR DESCRIPTION
**Pull request by akrantz**
The convertToSingleHost.js script will show usage if no arguments are given.

image

1. **Do these changes impact any *npm scripts* commands (in package.json)?** (e.g., running 'npm run start')
No

2. **Do these changes impact *VS Code debugging* options (launch.json)?**
No

3. **Do these changes impact *template output*?** (e.g., add/remove file, update file location, update file contents)
No

4. **Do these changes impact *documentation*?** (e.g., a tutorial on https://docs.microsoft.com/en-us/office/dev/add-ins/overview/office-add-ins)
No

If you answered yes to any of these please do the following:
    > Include 'Rick-Kirkham' in the review
    > Make sure the README file is correct

**Validation/testing performed**:

    `npm run convert-to-single-host` logs usage info to the console.

   `npm run convert-to-single-host powerpoint` converts the project for PowerPoint.

### CODE REVIEW

**Comment by millerds** [comment]
We should add information about which are required and what the defaults are for optionL arguments.

**Comment by Rick-Kirkham** [comment]
This logging is just for us, right? No external person is ever going to run this script at a command line, or are they?

**Comment by Rick-Kirkham** [comment]
> We should add information about which are required and what the defaults are for optionL arguments.

@akrantz Also, need to say that if the name has spaces in it, it has to be in quotation marks.

**Review by millerds — DISMISSED** [review]

File: convertToSingleHost.js [hunk 1]
```diff
@@ -4,12 +4,24 @@ const fs = require("fs");
...
 const childProcess = require("child_process");
+const hosts = ["excel", "onenote", "outlook", "powerpoint", "project", "word"];
+
+if (process.argv.length <= 2) {
+  const hostList = hosts.map((host) => `'${host}'`).join(", ");
+  console.log("SYNTAX: convertToSingleHost.js <host> <manifestType> <projectName> <appId>");
+  console.log();
+  console.log(`  host (required): Specifies which Office app will host the add-in: ${hostList}`);
+  console.log(`  manifestType: Specify the type of manifest to use: 'xml' or 'json'.  Defaults to 'xml'`);
+  console.log(`  projectName: The name of the project (use quotes when there are spaces in th name). Defaults to 'My Office Add-in'`);
```
Comment (Rick-Kirkham) [inline convertToSingleHost.js:15]:
```suggestion
  console.log(`  projectName: The name of the project (use quotes when there are spaces in the name). Defaults to 'My Office Add-in'`);
```

**Review by Rick-Kirkham — APPROVED** [review]
approving with one typo recommendation

**Review by millerds — APPROVED** [review]

File: convertToSingleHost.js [hunk 2]
```diff
@@ -4,12 +4,24 @@ const fs = require("fs");
...
 const util = require("util");
 const childProcess = require("child_process");
+const hosts = ["excel", "onenote", "outlook", "powerpoint", "project", "word"];
+
+if (process.argv.length <= 2) {
+  const hostList = hosts.map((host) => `'${host}'`).join(", ");
+  console.log("SYNTAX: convertToSingleHost.js <host> <manifestType> <projectName> <appId>");
+  console.log();
+  console.log(`  host (required): Specifies which Office app will host the add-in: ${hostList}`);
+  console.log(`  manifestType: Specify the type of manifest to use: 'xml' or 'json'.  Defaults to 'xml'`);
```
Comment (akrantz) [inline convertToSingleHost.js:14]: Consider "Default is" instead of "Defaults to" (for all lines). Content team can probably say what is best to use.

**Pull request merged**